### PR TITLE
Port refactors from legacy builder

### PR DESCRIPTION
**Pull request by QuentinI**
Port of https://github.com/EspressoSystems/hotshot-builder-core/pull/222

### CODE REVIEW

File: Cargo.toml
Comment (shenkeyao) [inline Cargo.toml:1]: We may consider updating to HotShot 0.5.67 without the `rc_` but this can be in a separate PR.

**Review by shenkeyao — APPROVED** [review]
LGTM!

File: Cargo.toml
Comment (shenkeyao) [reply Cargo.toml:1]: Oh actually, looks like the tag update is needed for conflict resolving.

Comment (QuentinI) [reply Cargo.toml:1]: Right, I'll bump the tag!

**Comment by QuentinI** [comment]
@shenkeyao merged main
